### DISCUSSION TITLE
How to set the homepage's preview title?

### DISCUSSION THREAD

**Discussion by boeboe26** [Kitware/trame#310] [Q&A] [answered]
I set the titles on the page, for example:
`state.trame__title ="My title`
`layout.title.set_text("My title")`

But when I share the link of the server, it takes a different one. It shows:

[trame is built by Kitware]
Application built with trame, a product from Kitware

How can I change these ones?

**Comment by jourdain** [comment, marked as answer]
That is a good one. Let me explain what is happening and how to change it.

So first of all trame rely on static HTML+JS but then once the server starts, the server will drive and update the content to match your expectation. In your use case, the link preview is performed on the static content with no dynamic server involved. 

So when you create a docker image to bundle your application, there is a step that build a "server" directory which contains a "server/www/index.html". Technically, you can edit that file before generating the final standalone image to provide any title/metadata/favicon/... that you like. But then to streamline the build/deployment process, you should take that file and put it inside the setup directory (`setup/www/index.html`) so that when the server directory get generated, that specific file get override by yours rather than the generic one from trame.

HTH
